### ISSUE TITLE
Reach Hacker News completeness

### ISSUE THREAD

**Issue by masak** [masak/007#335]
The ROADMAP is pretty good at laying out 007's short- and long-term goals. But I just realized there's an in-between goal post that we might call "Hacker News completeness".

When we showed 007 to HN back in 2015, the reaction of one commenter was ["no unquote?!"](https://news.ycombinator.com/item?id=10326221). Frankly I understand that reaction &mdash; 007 is sort of "the unquote language", in that unquotes set it apart from most other languages, and if they're not in place, 007 is all the worse for it.

So let's define "Hacker News completeness" as the small handful of pre-v1.0 things that would allow us to show 007 on Hacker News without getting a disappointed reaction akin to "no unquote?!". 😄 

Here's my list:

* `macro infix:<ff>` merged into master. (#207). Currently waiting on [displacing macro arguments](https://github.com/masak/007/issues/207#issuecomment-404487103).
* Unquotes. (#30). I mean, we _have_ unquotes, but the reason that issue hasn't closed yet is that unquotes dug up some deep parser difficulties (described [here](https://github.com/masak/007/issues/30#issuecomment-239562534)) that I believe #293 will fix. I expect #293 to be a _big_ undertaking, but it could also be angled so that we get promising feedback on it early.
* The *Quasis Are Lambdas* paradigm. I realized we didn't have a dedicated issue for it, so I just created #334 with links and explanation. This is related to #30, but could in principle be done before and independently.
* Good documentation/tutorial, specifically #67 and #295. I've started writing the text [in this gist](https://gist.github.com/masak/010ba2ab93a5632c1667155339ffd05b). It's a promising start, but there's ways to go yet. This goal is "soft" in the sense that it's "only" documentation; but I think it will make a big difference in practice for people seeing the language for the first time. And it's needed for v1.0 anyway.

I'd like to make this a goal for this summer. I think that's achievable.

"c'mon!" 😉

**Comment by masak** [comment]
> a goal for this summer

Heh. Reminds me of the Douglas Adams quote about deadlines. 😛 

Anyway. Since I've created this issue we've actually progressed on all four identified fronts, in small modest ways. So that's something, I guess.
